specification/authorization/resource-manager/Microsoft.Authorization/preview/2018-09-01-preview/authorization-RoleBasedCalls.json

```diff
@@ -1137,6 +1137,10 @@
           "type": "string",
           "description": "The principal ID assigned to the role. This maps to the ID inside the Active Directory. It can point to a user, service principal, or security group."
         },
+        "principalType": {
+          "type": "string",
+          "description": "The principal type of the assigned principal ID, e.g. user, service principal."
+        },
```

Comment:
Is there a list of known values for principal types?

Reply:
Yes, there is. Where should I include this?

Reply:
you can define it as an openapi enum type, like

```json
        "principalType": {
          "type": "string",
          "description": "The principal type of the assigned principal ID, e.g. user, service principal",
          "enum": [
            "User",
            "<other-value1>",
            "<other-value2>"
          ],
          "x-ms-enum": {
            "name": "PrincipalType",
            "modelAsString": true|false
          }
        }
```

List all the possible values under principalType::enum array.

If this list is limited and not going to change at least in this api-version then use `"modelAsString":false` otherwise `"modelAsString": true`.

Also is this a readonly property? then you can mark it as `readonly: true` as well.

Reply:
How can I update this PR?

Reply:
I've pushed the commit